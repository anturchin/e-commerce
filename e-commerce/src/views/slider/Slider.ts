import { View } from '../View';
import { ImgSlider } from './imgSlider/ImgSlider';
import { LinkPrev } from '../slidesShow/linkPrev/LinkPrev';
import { LinkNext } from '../slidesShow/linkNext/LinkNext';
import './Slider.scss';

export class Slider extends View {
    constructor() {
        super({ tag: 'div', classNames: ['mySlides', 'fade'] });
        this.setupSlider();
    }

    setupSlider() {
        const currentImg = new ImgSlider().getElement();
        this.viewHtmlElement.addInnerElement(currentImg);

        const prev = new LinkPrev().getElement();
        this.viewHtmlElement.addInnerElement(prev);

        const next = new LinkNext().getElement();
        this.viewHtmlElement.addInnerElement(next);
    }
}
