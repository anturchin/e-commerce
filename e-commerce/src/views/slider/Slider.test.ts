import { describe, expect, test } from '@jest/globals';
import { Slider } from './Slider';

describe('Slider', () => {
    test('should match', () => {
        const slider = new Slider();
        const renderedSlider = slider.getElement();
        expect(renderedSlider.outerHTML).toMatchSnapshot();
    });
});
