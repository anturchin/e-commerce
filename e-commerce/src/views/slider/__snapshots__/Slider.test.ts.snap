// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should match 1`] = `"<div class="mySlides fade"><img class="slider__img"><a class="prev">❮</a><a class="next">❯</a></div>"`;
